common/components/AddressFieldFactory/AddressFieldFactory.tsx

```diff
@@ -33,16 +48,36 @@ class AddressFieldFactoryClass extends React.Component<Props> {
     }
   }
 
+  public componentWillUnmount() {
+    if (this.goingToBlur) {
+      window.clearTimeout(this.goingToBlur);
+    }
+  }
+
   public render() {
+    const { isFocused } = this.state;
+
+    // Allow the click event on a dropdown item to process before hiding the dropdown.
+    const onBlur = () => (this.goingToBlur = window.setTimeout(this.blur, 150));
```

Comment:
This should be a class function, otherwise it'll always cause a re-render since it's a new function each time.

Reply:
:+1: